kademlia/src/main/java/ingsw/group1/kademlia/NodesRoutingTable.java

```diff
@@ -0,0 +1,169 @@
+package ingsw.group1.kademlia;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Extends Routing Table with {@link KBucket} that contains {@link Node<BinarySet>}
+ * Size of RoutingTable is fixed and equals to nodeOwner's key length
+ *
+ * @author Giorgia Bortoletti
+ */
+public class NodesRoutingTable extends RoutingTable<Node<BinarySet>, KBucket> {
+
+    private List<KBucket> bucketsTable;
+    private int sizeTable;
+    private PeerNode nodeOwner;
+
+    /**
+     * Constructor where the routing table length is equal to nodeOwner length
+     *
+     * @param nodeOwner {@link PeerNode} node owner of routing table
+     */
+    public NodesRoutingTable(PeerNode nodeOwner) {
+        new NodesRoutingTable(nodeOwner, nodeOwner.keyLength());
+    }
+
+    /**
+     * Constructor where the routing table length is sizeTable
+     *
+     * @param nodeOwner {@link PeerNode} node owner of routing table
+     * @param sizeTable dimension of routing table
+     */
+    public NodesRoutingTable(PeerNode nodeOwner, int sizeTable) {
+        this.nodeOwner = nodeOwner;
+        this.sizeTable = sizeTable;
+        bucketsTable = new ArrayList<>(sizeTable);
+        for (int i = 0; i < sizeTable; i++)
+            bucketsTable.add(new KBucket(sizeTable));
+    }
+
+    /**
+     * @return {@link PeerNode} that is node owner of this routing table
+     */
+    public PeerNode getNodeOwner() {
+        return nodeOwner;
+    }
+
+    /**
+     * @param node {@link Node<BinarySet>} to add
+     * @return true if the node has been added, false otherwise
+     */
+    @Override
+    public boolean add(Node<BinarySet> node) {
+        if (node == null)
+            return false;
+        return bucketsTable.get(getLocation(node)).add(node);
+    }
+
+    /**
+     * @param node {@link Node<BinarySet>} to remove
+     * @return true if the node has been removed, false otherwise
+     */
+    @Override
+    public boolean remove(Node<BinarySet> node) {
+        if (node == null)
+            return false;
+        int positionNode = getLocation(node);
+        KBucket bucketFound = bucketsTable.get(positionNode);
+        if (!bucketFound.contains(node))
+            return false;
+        return bucketFound.remove(node);
```

Comment:
I assumed it was removed because Java works with pointers. I'm sure that is removed from the KBucket.
For the table, I have added this `assertFalse(routingTable.contains(node));` after `remove` in the tests and the tests are verified.
Have you got other tips to further verify the removal?